lib/rspec/core/memoized_helpers.rb

```diff
@@ -250,6 +250,9 @@ def self.hook_intention
       # available to be called from within example group blocks.
       # You can think of them as being analagous to class macros.
       module ClassMethods
+        # @private
+        ReservedKeywords = %w[initialize run].freeze
```

Comment:
We definitely don't want to error on `run`.  There's no problem with users defining `let(:run)`.  For example, this works just fine:

``` ruby
RSpec.describe "Run" do
  let(:run) { 17 }

  specify { expect(run).to eq 17 }
  specify { expect(run).to eq 18 }
end
```

One spec passes and one fails, as you would expect.

I assume you are thinking of the `ExampleGroup.run` method, but that's a class method, not an instance method, so there is no conflict.  Some users have probably defined `let(:run)` in the wild and it would be a breaking change to give them errors for that when they upgrade.